### ISSUE TITLE
PHP Warnings when using the home page rules

### ISSUE THREAD

**Issue by marklchaves** [code-atlantic/content-control#106]
Getting the PHP warnings below when using the home page rules in a global restriction.

Watch this video for the steps to reproduce.

https://share.wppopupmaker.com/L1uyjXyR

### This warning appears for Content Is The Home Page and Content Is Not The Home Page 

```bash
Trying to get property 'ID' of non-object

    wp-content/plugins/content-control/inc/functions/rule-callbacks.php:88
    ContentControl\Rules\content_is_home_page()
    wp-content/plugins/content-control/inc/functions/rule-callbacks.php:88
    ContentControl\M\R\Rule->run_check()
    wp-content/plugins/content-control/classes/Models/RuleEngine/Rule.php:136
    ContentControl\M\R\Rule->check_rule()
    wp-content/plugins/content-control/classes/Models/RuleEngine/Query.php:101
    ContentControl\M\R\Query->check_rules()
    wp-content/plugins/content-control/classes/Models/RuleEngine/Group.php:71
    ContentControl\M\R\Group->check_rules()
    wp-content/plugins/content-control/classes/Models/RuleEngine/Query.php:103
    ContentControl\M\R\Query->check_rules()
    wp-content/plugins/content-control/classes/Models/Restriction.php:291
    ContentControl\Models\Restriction->check_rules()
    wp-content/plugins/content-control/classes/Services/Restrictions.php:215
    ContentControl\Services\Restrictions->get_applicable_restriction()
    wp-content/plugins/content-control/classes/Services/Restrictions.php:235
    ContentControl\Services\Restrictions->has_applicable_restrictions()
    wp-content/plugins/content-control/inc/functions/developers.php:31
    ContentControl\content_has_restrictions()
    wp-content/plugins/content-control/inc/functions/developers.php:61
    ContentControl\user_can_view_content()
    wp-content/plugins/content-control/inc/functions/developers.php:117
    ContentControl\content_is_restricted()
    wp-content/plugins/content-control/inc/functions/developers.php:280
    ContentControl\get_restriction_matches_for_queried_terms()
    wp-content/plugins/content-control/classes/Controllers/Frontend/Restrictions/QueryTerms.php:106
    ContentControl\C\F\R\QueryTerms->restrict_query_terms()
    wp-includes/class-wp-hook.php:324
    apply_filters('get_terms')
    wp-includes/taxonomy.php:1354
    get_terms()
    wp-includes/taxonomy.php:2291
    wp_get_object_terms()
    wp-includes/taxonomy.php:3804
    update_object_term_cache()
    wp-includes/post.php:7804
    _prime_post_caches()
    wp-content/plugins/woocommerce/includes/class-wc-cart-session.php:126
    WC_Cart_Session->get_cart_from_session()
    wp-includes/class-wp-hook.php:324
    do_action('wp_loaded')
    wp-settings.php:665
```

### The following 2 warnings happen for Content Is Not The Home Page 

```bash
Trying to get property 'post_title' of non-object	

    wp-content/plugins/content-control/classes/QueryMonitor/Output.php:219
    ContentControl\QueryMonitor\Output->output_post_restrictions()
    wp-content/plugins/content-control/classes/QueryMonitor/Output.php:113
    ContentControl\QueryMonitor\Output->output()
    wp-content/plugins/query-monitor/dispatchers/Html.php:326
    do_action('shutdown')
    wp-includes/load.php:1260
    shutdown_action_hook()
    wp-includes/load.php:1260
```

```bash
Trying to get property 'post_type' of non-object	

    wp-content/plugins/content-control/classes/QueryMonitor/Output.php:220
    ContentControl\QueryMonitor\Output->output_post_restrictions()
    wp-content/plugins/content-control/classes/QueryMonitor/Output.php:113
    ContentControl\QueryMonitor\Output->output()
    wp-content/plugins/query-monitor/dispatchers/Html.php:326
    do_action('shutdown')
    wp-includes/load.php:1260
    shutdown_action_hook()
    wp-includes/load.php:1260
```

**Comment by danieliser** [comment]
Should be resolved in a89b1a8

**Comment by marklchaves** [comment]
Confirmed! Thank you 🤩

> Should be resolved in [a89b1a8](https://github.com/code-atlantic/content-control/commit/a89b1a86e2bb79f52c0db54be62a71e923c8fba5)
